### PR TITLE
Update init method for storm_europe

### PR DESCRIPTION
**Pull request by zeliest**
Changes proposed in this PR:
- Update init method for storm_europe

This PR fixes issue #

### PR Author Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] Correct target branch selected (if unsure, select `develop`)
- [ ] Source branch up-to-date with target branch
- [ ] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: /CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

File: climada/hazard/storm_europe.py [hunk 1]
```diff
@@ -81,13 +82,18 @@ class StormEurope(Hazard):
...
 
-    def __init__(self):
+    def __init__(self,
+                 units: Optional[str] = 'm/s',
+                 ssi: Optional[np.ndarray] = None,
+                 ssi_wisc: Optional[np.ndarray] = None,
+                 ssi_full_area: Optional[np.ndarray] = None
+                 ):
         """Calls the Hazard init dunder. Sets unit to 'm/s'."""
         Hazard.__init__(self, HAZ_TYPE)
```
Comment (emanuel-schmid) [inline climada/hazard/storm_europe.py:92]: parameter description is missing in pydoc string

Comment (peanutfun) [reply climada/hazard/storm_europe.py:108]: I added a bit, but I need some help. I have no idea what the attributes mean and how they are supposed to be shaped 🙈 @chahank @tovogt, maybe you can help?

File: climada/hazard/storm_europe.py [hunk 2]
```diff
@@ -814,9 +836,8 @@ def generate_prob_storms(self, reg_id=528, spatial_shift=4, ssi_args=None,
                     **kwargs)
 
         LOGGER.info('Generating new StormEurope instance')
-        new_haz = StormEurope()
+        new_haz = StormEurope(ssi=ssi)
         new_haz.intensity = sparse.csr_matrix(intensity_prob)
-        new_haz.ssi_full_area = ssi
```
Comment (tovogt) [inline climada/hazard/storm_europe.py:841]: Make sure to use `ssi=ssi_full_area` here instead of `ssi=ssi`!

Comment (tovogt) [reply climada/hazard/storm_europe.py]: Thanks @emanuel-schmid for dealing with this!

File: climada/hazard/storm_europe.py (hunk 1 above)
Comment (tovogt) [reply climada/hazard/storm_europe.py]: In order to finalize the docstrings, we will probably need input from somebody who is familiar with this hazard type...

Comment (emanuel-schmid) [reply climada/hazard/storm_europe.py]: True. I hope @mmyrte can help here. 🙏

**Review by mmyrte — APPROVED** [review]
I was first going "oh my, this is fancy, passing references to output arrays", but when I saw what I had done further down, it made sense. I'm happy with the changes, thanks!

File: climada/hazard/storm_europe.py [hunk 3]
```diff
@@ -81,13 +82,32 @@ class StormEurope(Hazard):
...
+            The Storm Severity Index (SSI). Shape ``(?? x ??)``. Defaults to an empty
+            array.
+        ssi_wisc : numpy.ndarray
+            The Storm Severity Index (SSI) as reconded in the footprint files.
+            Shape ``(?? x ??)``. Defaults to an empty array.
+        ssi_full_area : numpy.ndarray
+            The Storm Severity Index (SSI) ???. Shape ``(?? x ??)``. Defaults to an empty
+            array.
+        """
+        Hazard.__init__(self, haz_type=HAZ_TYPE, units=units, **kwargs)
```
Comment (bguillod) [inline climada/hazard/storm_europe.py:107]: Actually, why not `Super.__init__` here? There might be a good reason so I'm curious.

Comment (emanuel-schmid) [reply climada/hazard/storm_europe.py:107]: No particular reason I guess. `super()` would be cleaner: you could change the super class by changing the one line of code that declares the class. But I don't expect `StormEurope` to inherit from anything else then `Hazard` ever.

**Comment by mmyrte** [comment]
> @mmyrte Can you please have a look at the docstring of the `__init__` above and suggest better descriptions for the parameters?

@tovogt Sorry, was otherwise occupied. The unit tests indicate that the SSI is a 1d vector of equal length as the number of events. I'll check out the branch and ammend those lines, just a moment.

File: climada/hazard/storm_europe.py [hunk 4]
```diff
@@ -81,13 +82,35 @@ class StormEurope(Hazard):
...
+            The units of the Hazard intensity. Defaults to 'm/s'
+        ssi : numpy.ndarray
+            The Storm Severity Index (SSI). 1d vector, same length as number of
+            storms. Defaults to an empty array.
+        ssi_wisc : numpy.ndarray
+            The Storm Severity Index (SSI) as reconded in the footprint files.
+            1d vector, same length as number of storms. Defaults to an empty
+            array.
+        ssi_full_area : numpy.ndarray
+            FIXME need to recover rationale for this attribute; no substantive
```
Comment (mmyrte) [inline climada/hazard/storm_europe.py:105]: @ThomasRoosli do you recall why we had this additional ssi_full_area thing in here? It's only ever used when computing a synthetic event set. I seem to remember that there was a good reason to not set the synthetic SSI to the same attribute name present in the reanalysis/WISC synthetic data. However, I don't remember, and we seem not to have documented it. I propose to simply drop this additional parameter. I would say this PR is as good an occasion as any.

**Review by mmyrte — CHANGES_REQUESTED** [review]

File: climada/hazard/storm_europe.py [hunk 5]
```diff
@@ -159,27 +183,30 @@ def from_footprints(cls, path, description=None, ref_raster=None, centroids=None
...
-        for file_name in file_names:
-            if any(fo in file_name for fo in files_omit):
-                LOGGER.info("Omitting file %s", file_name)
-                continue
-            new_haz = cls._read_one_nc(file_name, centroids, intensity_thres)
-            if new_haz is not None:
-                haz.append(new_haz)
-
+        files_to_read = [file for file in file_names if file not in files_omit]
+        LOGGER.info(
```
Comment (emanuel-schmid) [inline climada/hazard/storm_europe.py:187]:
```suggestion
        if files_omit: LOGGER.info(
```

File: climada/hazard/storm_europe.py [hunk 6]
```diff
@@ -159,27 +183,30 @@ def from_footprints(cls, path, description=None, ref_raster=None, centroids=None
...
+            "Omitting files %s", [
+                file for file in file_names if file in files_omit])
```
Comment (emanuel-schmid) [inline climada/hazard/storm_europe.py:189]:
```suggestion
        if any(fo in file_name for fo in files_omit):
            LOGGER.info("Omitting files %s", [
                file for file in file_names if file in files_omit])
```

**Comment by emanuel-schmid** [comment]
@peanutfun Good catch! 🙌 Also nice, to leave the fraction as `None` while we're at it. 😀

File: climada/hazard/storm_europe.py (hunk 6 above)
Comment (peanutfun) [reply climada/hazard/storm_europe.py]: Very important suggestion, thank you! But I did not want to add yet another list comprehension. I found a different solution using sets, see 6f0acc1ce498a59090fd8669f68eda50a928cb44

File: climada/hazard/storm_europe.py (hunk 4 above)
Comment (ThomasRoosli) [reply climada/hazard/storm_europe.py:105]: I would have to look closer into the code. From my memory: `ssi_full_area` is needed to conserve the index that is derived from the area and windspeed of the full geographic extend of a storm system. This is importent if a StormEurope object is cropped to a smaller region (e.g. a country).

Comment (emanuel-schmid) [reply climada/hazard/storm_europe.py:105]: @ThomasRoosli: thanks for chipping in! In this case I'd be fine to keep the attribute and write just that into the description:
```
   Used to conserve the the index that is derived from the area and windspeed 
   of the full geographic extend of a storm system. Needed for cropping the 
   `StormEurope` object into a smaller region`
```

File: climada/hazard/storm_europe.py (hunk 6 above)
Comment (emanuel-schmid) [reply climada/hazard/storm_europe.py]: Rather very unimportant, right? But yes, sets make more sense. 👍 (I was thinking along the same lines. Let's just hope the list was rather a bug than e feature, i.e., providing a path list with duplicates is not an intended valid use case. 🤞)

File: climada/hazard/storm_europe.py [hunk 7]
```diff
@@ -81,13 +82,36 @@ class StormEurope(Hazard):
...
+            storms. Defaults to an empty array.
+        ssi_wisc : numpy.ndarray
+            The Storm Severity Index (SSI) as reconded in the footprint files.
+            1d vector, same length as number of storms. Defaults to an empty
+            array.
+        ssi_full_area : numpy.ndarray
+            FIXME need to recover rationale for this attribute; no substantive
+            difference to regular SSI attribute; is computed when creating a
+            synthetic event set. 1d vector, same length as number of storms.
+            Defaults to an empty array.
```
Comment (emanuel-schmid) [inline climada/hazard/storm_europe.py:108]:
```suggestion
            1d vector, same length as number of storms.
            Used to conserve the the index that is derived from the area and windspeed 
            of the full geographic extend of a storm system. Needed for cropping the 
            `StormEurope` object into a smaller region`.
            Defaults to an empty array.
```

File: climada/hazard/storm_europe.py [hunk 8]
```diff
@@ -81,13 +82,37 @@ class StormEurope(Hazard):
...
+            The Storm Severity Index (SSI). 1d vector, same length as number of
+            storms. Defaults to an empty array.
+        ssi_wisc : numpy.ndarray
+            The Storm Severity Index (SSI) as reconded in the footprint files.
+            1d vector, same length as number of storms. Defaults to an empty
+            array.
+        ssi_full_area : numpy.ndarray
+            1d vector, same length as number of storms.
+            Used to conserve the the index that is derived from the area and windspeed 
+            of the full geographic extend of a storm system. Needed for cropping the
```
Comment (emanuel-schmid) [inline climada/hazard/storm_europe.py:107]:
```suggestion
            of the full geographic extent of a storm system. Needed for cropping the 
```

**Pull request merged**